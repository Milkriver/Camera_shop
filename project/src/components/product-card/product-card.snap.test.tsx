import {render} from '@testing-library/react';
import { createMemoryHistory } from 'history';
import HistoryRouter from '../history-router/history-router';
import ProductCard from './product-card';

const history = createMemoryHistory();
const mockOffer = {
    id: 1,
    name: '',
    vendorCode: '',
    type: '',
    category: '',
    description: '',
    level: '',
    rating: 1,
    price: 1,
    previewImg: '',
    previewImg2x: '',
    previewImgWebp: '',
    previewImgWebp2x: '',
    reviewCount: 1,
  };

describe('Component: ProductCard', () => {
  it('should render correctly', () => {
    const {container} = render(
        <HistoryRouter history={history}>
            <ProductCard isActive={false} product={mockOffer}/>
      </HistoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});