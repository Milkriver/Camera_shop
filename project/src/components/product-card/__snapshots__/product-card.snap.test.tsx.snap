// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ProductCard should render correctly 1`] = `
<div>
  <div
    class="product-card "
  >
    <div
      class="product-card__img"
    >
      <picture>
        <source
          srcset="/"
          type="/"
        />
        <img
          alt=""
          height="240"
          src="/"
          srcset="/"
          width="280"
        />
      </picture>
    </div>
    <div
      class="product-card__info"
    >
      <div
        class="rate product-card__rate"
      >
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-full-star"
          />
        </svg>
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
        <p
          class="visually-hidden"
        >
          Рейтинг: 
          1
        </p>
        <p
          class="rate__count"
        >
          <span
            class="visually-hidden"
          >
            Всего оценок:
          </span>
          1
        </p>
      </div>
      <p
        class="product-card__title"
      />
      <p
        class="product-card__price"
      >
        <span
          class="visually-hidden"
        >
          Цена:
        </span>
        1
         ₽
      </p>
    </div>
    <div
      class="product-card__buttons"
    >
      <button
        class="btn btn--purple product-card__btn"
        type="button"
      >
        Купить
      </button>
      <a
        class="btn btn--transparent"
        href="/product/1"
      >
        Подробнее
      </a>
    </div>
  </div>
</div>
`;
